src/dsp_tools/models/connection.py

```diff
@@ -27,240 +29,260 @@ def check_for_api_error(response: requests.Response) -> None:
-        return self._token
+        self.token = result["token"]
 
     def start_logging(self) -> None:
         """Start writing every API call to a file"""
-        self._log = True
+        self.log = True
+        self.log_directory.mkdir(exist_ok=True)
 
-    def stop_logging(self):
+    def stop_logging(self) -> None:
         """Stop writing every API call to a file"""
-        self._log = False
+        self.log = False
 
     def logout(self) -> None:
-        """
-        Performs a logout
-        :return: None
-        """
-
-        if self._token is not None:
+        """Delete the session token"""
+        if self.token:
             response = requests.delete(
-                self._server + "/v2/authentication",
-                headers={"Authorization": "Bearer " + self._token},
+                self.server + "/v2/authentication",
+                headers={"Authorization": "Bearer " + self.token},
                 timeout=20,
             )
             check_for_api_error(response)
-            self._token = None
-
-    def __del__(self):
-        pass
-        # self.logout()
-
-    def post(self, path: str, jsondata: Optional[str] = None):
+            self.token = None
+
+    def write_log_file(
+        self,
+        method: str,
+        route: str,
+        headers: dict[str, str],
+        jsondata: Optional[str],
+        params: Optional[dict[str, Any]],
+        response: requests.Response,
+    ) -> None:
         """
-        Post Json data to a given server using a HTTP POST request
-        :param path: Path of RESTful route
-        :param jsondata: Valid JSON as string
-        :return: Response from server
+        If logging is enabled (self.log == True), write the request and response to a file.
+
+        Args:
+            method: HTTP method (POST, GET, PUT, DELETE)
+            route: route of DSP-API that was called
+            headers: headers of the HTTP request
+            jsondata: data sent to the server
+            response: response of the server
         """
+        if not self.log:
+            return
+        if response.status_code == 200:  # pylint: disable=duplicate-code
+            _return = response.json()
+        else:
+            _return = {"status": response.status_code, "message": response.text}
+        logobj = {
+            "DSP server": self.server,
+            "route": route,
+            "method": method,
+            "headers": headers,
+            "params": params,
+            "body": json.loads(jsondata) if jsondata else None,
+            "return-headers": dict(response.headers),
+            "return": _return,
+        }
+        filename = f"{datetime.now().strftime('%Y-%m-%d %H.%M.%S.%f')} {method} {route.replace('/', '_')}.json"
+        with open(self.log_directory / filename, "w", encoding="utf-8") as f:
+            json.dump(logobj, f, indent=4)
+
+    def post(
+        self,
+        path: str,
+        jsondata: Optional[str] = None,
```

Comment:
how can this be none?

Reply:
There are quite some examples in the codebase where it is none, e.g. here: https://github.com/dasch-swiss/dsp-tools/blob/wip/dev-2534-refactor-models/src/dsp_tools/models/user.py#L543

Reply:
oof... bad API design from our side! 🙈